contracts/vortex/CarbonVortex.sol

```diff
@@ -3,58 +3,102 @@ pragma solidity 0.8.19;
-}
+import { ExpDecayMath } from "../utility/ExpDecayMath.sol";
+import { PPM_RESOLUTION, MAX_GAP } from "../utility/Constants.sol";
 
 /**
  * @dev CarbonVortex contract
  */
 contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable, Utils {
     using Address for address payable;
+    using SafeCast for uint256;
+
+    uint128 private constant INITIAL_PRICE_SOURCE_AMOUNT = type(uint128).max;
+    uint128 private constant INITIAL_PRICE_TARGET_AMOUNT = 1e12;
 
+    // addresses for token withdrawal
     ICarbonController private immutable _carbonController;
-    IBancorNetwork private immutable _bancorNetwork;
-    Token private immutable _bnt;
+    IVault private immutable _oldVortex;
+    IVault private immutable _vault;
+
+    // address for token collection - collects all swapped target/final target tokens
+    address payable private immutable _transferAddress;
 
-    uint256 private _totalBurned;
+    // first token for swapping
+    Token private immutable _targetToken;
+    // second (optional) token for swapping
+    Token private immutable _finalTargetToken;
+
+    // total target (if no finalTarget token is defined) / finalTarget tokens collected in transferAddress
+    uint256 private _totalCollected;
 
     // rewards percentage
-    uint256 private _rewardsPPM;
+    uint32 private _rewardsPPM;
+
+    // price reset multiplier - used to reset the price after a trade in special cases
+    uint32 private _priceResetMultiplier;
+
+    // min token sale amount multiplier - used to reset the price after execute in special cases
+    uint32 private _minTokenSaleAmountMultiplier;
+
+    // time until the price gets halved for the target token during a trade
+    uint32 private _targetTokenPriceDecayHalfLife;
+
+    // time until the price gets halved for the target token on price reset during a trade
+    uint32 private _targetTokenPriceDecayHalfLifeOnReset;
+
+    // time until the price gets halved for all tokens when auction is initialized
+    uint32 private _priceDecayHalfLife;
+
+    // token to pair disabled mapping (disabled pairs aren't tradeable)
+    mapping(Token token => bool pairDisabled) private _disabledPairs;
+
+    // token to trading start time mapping
+    mapping(Token token => uint32 tradingStartTime) private _tradingStartTimes;
+
+    // token to initial price mapping
+    mapping(Token token => Price initialPrice) private _initialPrice;
+
+    // min token sale amounts - resets the token price if below this amount after a call to execute
+    // resets the current sale amount if below this amount after a trade (for target token)
+    mapping(Token token => uint128 _minTokenSaleAmount) private _minTokenSaleAmounts;
+
+    // initial and current target token sale amount - for targetToken->finalTargetToken trades
+    SaleAmount private _targetTokenSaleAmount;
 
     // upgrade forward-compatibility storage gap
-    uint256[MAX_GAP - 2] private __gap;
+    uint256[MAX_GAP - 7] private __gap;
```

Comment:
I think that it should be 8. Did you verify by printing the storage layout? Also, the forge suit doesn't have an upgradeable storage verification test, like we had in hardhat, right?